File: FSUIKit/Classes/FSSwitchCell.m
////
////  FSSwitchCell.m
////  Expand
////
////  Created by Fudongdong on 2017/11/29.
////  Copyright © 2017年 china. All rights reserved.
////
//
//#import "FSSwitchCell.h"
//
//@implementation FSSwitchCell {
//    UISwitch    *_s;
//}
//
//- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
//    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
//    if (self) {
//        [self switchDesignViews];
//    }
//    return self;
//}
//
//- (void)switchDesignViews {
//    self.userInteractionEnabled = YES;
//    self.selectionStyle = UITableViewCellSelectionStyleNone;
//    _s = [[UISwitch alloc] initWithFrame:CGRectMake([UIScreen mainScreen].bounds.size.width - 71, 12, 51, 31)];
//    [_s addTarget:self action:@selector(switchAction:) forControlEvents:UIControlEventValueChanged];
//    [self.contentView addSubview:_s];
//}
//
//- (void)switchAction:(UISwitch *)s {
//    _on = s.isOn;
//    if (self.block) {
//        self.block(s);
//    }
//}
//
//- (void)setOn:(NSInteger)on {
//    _s.on = on;
//}
//
//@end

File: FSUIKit/Classes/Swift/FSSwitchCell.swift
// FSSwitchCellS.swift
// Translated from FSSwitchCell.h/m

import UIKit

@objcMembers
public class FSSwitchCell: UITableViewCell {
    
    private var _switch: UISwitch!
    
    public var on: Int {
        get { _switch.isOn ? 1 : 0 }
        set { _switch.isOn = newValue != 0 }
    }
    
    public var block: ((UISwitch) -> Void)?
    
    public override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        switchDesignViews()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        switchDesignViews()
    }
    
    private func switchDesignViews() {
        isUserInteractionEnabled = true
        selectionStyle = .none
        _switch = UISwitch(frame: CGRect(x: UIScreen.main.bounds.width - 71, y: 12, width: 51, height: 31))
        _switch.addTarget(self, action: #selector(switchAction(_:)), for: .valueChanged)
        contentView.addSubview(_switch)
    }
    
    @objc private func switchAction(_ s: UISwitch) {
        block?(s)
    }
}
